app/controllers/collections_controller.rb

```diff
@@ -8,4 +8,26 @@ def form_class
   def presenter_class
     CollectionShowPresenter
   end
+
+  def manifest
+    respond_to do |f|
+      f.json do
+        render json: manifest_builder
+      end
+    end
+  end
+
+  private
+
+    def manifest_builder
+      ManifestBuilder.new(presenter, ssl: request.ssl?, services: AuthManifestBuilder.auth_services(login_url, logout_url))
+    end
+
+    def login_url
+      main_app.user_omniauth_authorize_url(:cas)
```

Comment:
I think the whole builder infrastructure needs redesigned to allow for more flexibility to handle that kind of stuff. The churn in this file is kind of unreal.